.github/workflows/build.yaml

```
@@ -0,0 +1,57 @@
+name: Build Docker image and push to Dockerhub
+
+on:
+  push:
+    branches:
+      - "main"
+env:
+  IMAGE_NAME: stringerrss/stringer
+
+jobs:
+  buildx:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Extract Docker sha tag
+        id: get-tag-sha
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.IMAGE_NAME }}
+          tags: type=sha
+
+      - name: Extract Docker latest tag
+        id: get-tag-latest
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.IMAGE_NAME }}
+          tags: type=raw, value=latest
+
+      - name: Log in to Docker Hub
```

If we're building this on every branch, I think we'll need to skip this step unless we're on `main`:

```suggestion
      - name: Log in to Docker Hub
        if: ${{ github.ref_name == 'main' }}
```